Sources/FunctionalKit/FutureType.swift

```diff
@@ -13,7 +13,7 @@ public protocol FutureType: TypeConstructor {
 public final class Future<A>: FutureType {
 	public typealias ParameterType = A
 
-	private enum State<T> {
+	private indirect enum State<T> {
```

Comment:
Why is that?

Reply:
This is because a Swift bug, but a better solution is to extract the enum and make it fileprivate

Reply:
Yep, we did like that in another branch we're working on to add tests for functor, applicative and monads laws on all types. We can leave it `indirect` for now, we'll fix the conflict later :)